src/titiler/xarray/titiler/xarray/io.py

```diff
@@ -0,0 +1,273 @@
+        xr_open_args["engine"] = "zarr"
+        xr_open_args["consolidated"] = consolidated
+
+    # Additional arguments when dealing with a reference file.
+    if reference:
+        xr_open_args["consolidated"] = False
+        xr_open_args["backend_kwargs"] = {"consolidated": False}
+
+    ds = xarray.open_dataset(file_handler, **xr_open_args)
+
+    if cache_client:
+        # Serialize the dataset to bytes using pickle
+        data_bytes = pickle.dumps(ds)
+        cache_client.set(cache_key, data_bytes)
+
+    return ds
+
+
+def arrange_coordinates(da: xarray.DataArray) -> xarray.DataArray:
+    """Arrange coordinates to DataArray.
+
+    An rioxarray.exceptions.InvalidDimensionOrder error is raised if the coordinates are not in the correct order time, y, and x.
+    See: https://github.com/corteva/rioxarray/discussions/674
+
+    We conform to using x and y as the spatial dimension names. You can do this a bit more elegantly with metpy but that is a heavy dependency.
+
+    """
+    if "x" not in da.dims and "y" not in da.dims:
+        latitude_var_name = "latitude" if "latitude" in da.dims else "lat"
+        longitude_var_name = "longitude" if "longitude" in da.dims else "lon"
+        da = da.rename({latitude_var_name: "y", longitude_var_name: "x"})
+
+    if "time" in da.dims:
+        da = da.transpose("time", "y", "x")
+    else:
+        da = da.transpose("y", "x")
+
+    return da
+
+
+def get_variable(
+    ds: xarray.Dataset,
+    variable: str,
+    datetime: Optional[str] = None,
+    drop_dim: Optional[str] = None,
+) -> xarray.DataArray:
+    """Get Xarray variable as DataArray."""
+    da = ds[variable]
+    da = arrange_coordinates(da)  # TODO: Duplicated call L173
+
+    # TODO: add test
+    if drop_dim:
+        dim_to_drop, dim_val = drop_dim.split("=")
+        da = da.sel({dim_to_drop: dim_val}).drop(dim_to_drop)
+
+    da = arrange_coordinates(da)
+
+    # Make sure we have a valid CRS
+    crs = da.rio.crs or "epsg:4326"
+    da = da.rio.write_crs(crs)
+
+    if crs == "epsg:4326" and (da.x > 180).any():
+        # Adjust the longitude coordinates to the -180 to 180 range
+        da = da.assign_coords(x=(da.x + 180) % 360 - 180)
+
+        # Sort the dataset by the updated longitude coordinates
+        da = da.sortby(da.x)
+
+    if "time" in da.dims:
+        if datetime:
+            time_as_str = datetime.split("T")[0]
+            if da["time"].dtype == "O":
+                da["time"] = da["time"].astype("datetime64[ns]")
+            da = da.sel(
+                time=numpy.array(time_as_str, dtype=numpy.datetime64), method="nearest"
+            )
+        else:
+            da = da.isel(time=0)
+
+    return da
+
+
+@attr.s
+class Reader(XarrayReader):
+    """Reader: Open Zarr file and access DataArray."""
+
+    src_path: str = attr.ib()
+    variable: str = attr.ib()
+
+    # xarray.Dataset options
+    opener: Callable[..., xarray.Dataset] = attr.ib(default=xarray_open_dataset)
```

Comment:
By making the `opener` an option it eases the user to bring their own `xarray_open_dataset` callable which may add more `protocol` (e.g Azure, GCS ...)